server/utils/auth/command_gate.py

```
@@ -0,0 +1,365 @@
+        return False
+    try:
+        from utils.db.connection_pool import db_pool
+        with db_pool.get_user_connection() as conn:
+            cursor = conn.cursor()
+            cursor.execute(
+                "SELECT security_tainted FROM chat_sessions WHERE id = %s",
+                (session_id,),
+            )
+            row = cursor.fetchone()
+            return bool(row and row[0])
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)